Map/Test/MapTest.cpp

```diff
@@ -1,10 +1,30 @@
+#include "FieldTypes.hpp"
+#include "SquareMap.hpp"
 #include <gtest/gtest.h>
-#include "Map.hpp"
 
 using namespace ::testing;
 
-// TEST(Maptest, GetRoomsShouldReturnOneRoom)
-// {
-//     Map map(1);
-//     ASSERT_EQ(map.getRooms().size(), 1);
-// }
\ No newline at end of file
+TEST(Maptest, quantityOfMapColumnsShouldEqualMapSize)
+{
+    constexpr auto mapSize = 8u;
+    SquareMap map(mapSize);
+
+    ASSERT_EQ(map.getFields().size(), mapSize);
+}
+
+TEST(Maptest, quantityOfFieldsInFirstColumnShouldEqualMapSize)
+{
+    constexpr auto mapSize = 8u;
+    SquareMap map(mapSize);
```

Comment:
Dobrze Ci idzie, dlatego chcę wprawadzić nową rzecz, test fixture. Przydatna rzecz do testowania klas.

class SquareMapTest : public Test
{
public:
   SquareMapTest() : _sut (mapSize){}
   SquareMap _sut; // sut - system under test
}
TEST_F(nazwaKlasy, nazwa testu)
TEST_F(SquareMapTest, quantityOfFieldsInFirstColumnShouldEqualMapSize)
{
ASSERT_EQ(_sut.getFields().at(0).size(), mapSize);
}

Teraz nie musimy za każdym razem inicjalizować SquareMap w każdym teście, bo mamy dostęp do zmiennych które zdefiniowaliśmy w klasie SquareMapTest.

Jak masz do tego pytania, bo to nowa rzecz, to pytaj na slacku.